From e8a86b273d7cef26961c12c1441c57d2f9046134 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Sat, 12 Oct 2019 20:09:06 +0200
Subject: [PATCH] tests: work around the "MSYS vs Windows paths" problem

The tests fail for the wrong reason: they cannot even access the files
because their paths are given in MSYS format (because the test script is
written in Perl and is executed by MSYS2's Perl interpreter), but
`openssl.exe` is a MINGW program, so it needs Windows paths.

There are probably more elegant ways to achieve the same, but I really
only need to get the tests to run with the correct paths, and I cannot
find anything on the web that suggests how to do this without forking
and executing `cygpath.exe`. So I give up on a better way, for now.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 util/perl/OpenSSL/Test.pm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/perl/OpenSSL/Test.pm b/util/perl/OpenSSL/Test.pm
index 49fb88f..dccefd0 100644
--- a/util/perl/OpenSSL/Test.pm
+++ b/util/perl/OpenSSL/Test.pm
@@ -871,7 +871,11 @@ sub __env {
     (my $recipe_datadir = basename($0)) =~ s/\.t$/_data/i;
 
     $directories{SRCTOP}  = abs_path($ENV{SRCTOP} || $ENV{TOP});
+    $directories{SRCTOP}  = `cygpath -m $directories{SRCTOP}`;
+    $directories{SRCTOP} =~ s/\r?\n$//;
     $directories{BLDTOP}  = abs_path($ENV{BLDTOP} || $ENV{TOP});
+    $directories{BLDTOP}  = `cygpath -m $directories{BLDTOP}`;
+    $directories{BLDTOP} =~ s/\r?\n$//;
     $directories{BLDAPPS} = $ENV{BIN_D}  || __bldtop_dir("apps");
     $directories{SRCAPPS} =                 __srctop_dir("apps");
     $directories{BLDFUZZ} =                 __bldtop_dir("fuzz");
-- 
2.23.0.windows.1

